libcalico-go/lib/backend/model/keys_test.go

```diff
@@ -15,17 +15,137 @@
+	"/calico/v1/host/foobar/endpoint",
+	"/calico/v1/host/foobar/endpoint/biff",
+	"/calico/v1/host/foobar/metadata",
+	"/calico/v1/host/foobar/wireguard",
+	"/calico/v1/host/foobar/config/biff/bopp",
+	"/calico/v1/host/foobar/bird_ip",
+	"/calico/v1/host/foobar",
+	"/calico/v1/netset",
+	"/calico/v1/netset/foo",
+	"/calico/v1/Ready",
+	"/calico/v1/Ready/garbage",
+	"/calico/v1/policy/tier",
+	"/calico/v1/policy/tier/foo",
+	"/calico/v1/policy/tier/foo/policy",
+	"/calico/v1/policy/tier/foo/policy/bar",
+	"/calico/v1/policy/profile",
+	"/calico/bgp/v1/global",
+	"/calico/bgp/v1/global/peer_v4",
+	"/calico/bgp/v1/global/peer_v4/name",
+	"/calico/bgp/v1/global/peer_v4/name",
+	"/calico/bgp/v1/global/peer_v4/10.0.0.5",
+	"/calico/bgp/v1/global/peer_v4/10.0.0.5-500",
+	"/calico/bgp/v1/global/peer_v6",
+	"/calico/bgp/v1/global/peer_v6/name",
+	"/calico/bgp/v1/host",
+	"/calico/bgp/v1/host/peer_v4",
+	"/calico/bgp/v1/host/peer_v4/name",
+	"/calico/bgp/v1/host/peer_v6",
+	"/calico/bgp/v1/host/peer_v6/name",
+	"/calico/bgp/v1/host/random-node-2/peer_v6/aabb:aabb::ffff-123",
+	"/calico/v1",
+	"/calico/v1/ipam",
+	"/calico/ipam/v2/assignment/ipv4/1.2.3.4-26",
+	"/calico/ipam/v2/handle/foobar",
+	"/calico/ipam/v2/handle/foobar/baz",
+	"/calico/ipam/v2",
+	"/calico/ipam/v2/host/foobar/ipv4/block/1.2.3.4-5",
+	"calico/ipam/v2/host/0/ipv0/block/0",
+	"/calico/felix/v1/host",
+	"/calico/felix/v1/host/foo/endpoint/bar",
+	"/calico/felix/v1/endpoint",
+	"/calico/felix/v2/foo/host",
+	"/calico/felix/v2//foo/host",
+	"/calico/felix/v2/region-Europe/host/h1/status",
+	"/calico/resources/v3/projectcalico.org/foo/bar/baz",
+	"/bar/v1/host/foobar/workload/open%2fstack/work%2fload/endpoint/end%2fpoint",
+	"/calico/v1/host/foobar/workload/open%2fstack/work%2fload/endpoint/end%2fpoint",
+	"/calico/resources/v3/projectcalico.org/felixconfigurations/default",
+	"/calico/resources/v3/projectcalico.org/networkpolicies",
+	"/calico/resources/v3/projectcalico.org/networkpolicies/default/my-network-policy",
+	"/calico/resources/v3/projectcalico.org/felixconfigurations/default/my-network-policy",
+	"/calico/felix/v2/region-Europe/host/h1/workload/o1/w1/endpoint/e1",
+}
+
+// FuzzKeyFromDefaultPath fuzzes KeyFromDefaultPath, comparing its output and behaviour with the
+// older implementation.
+func FuzzKeyFromDefaultPath(f *testing.F) {
+	for _, k := range interestingPaths {
+		f.Add(k)
+		f.Add(k[1:])
+		f.Add(k + "/")
+		f.Add("/" + k + "//")
+		f.Add("/cluster/cluster-1" + k)
+	}
+	f.Fuzz(func(t *testing.T, path string) {
+		oldKey := OldKeyFromDefaultPath(path)
+		newKey := KeyFromDefaultPath(path)
+		if oldKey == nil {
+			// Old version couldn't parse this, so we don't care if the new one can.
+			return
+		}
+		if newKey == nil {
+			t.Errorf("Old KeyFromDefaultPath(%q)=%v but new KeyFromDefaultPath() returned nil", path, oldKey)
+			return
+		}
+
+		if !safeKeysEqual(oldKey, newKey) {
+			t.Errorf("%q -> (new output) %v != (old output) %v", path, newKey, oldKey)
```

Comment:
`Errorf` doesn't return on its own.  I'll swap it for `Fatalf`, which terminates the test.

(I was only returning in some places because those places were guarding further logic.)

Reply:
But now we have only `Fatalf`s, so all of the cases are terminating, IIUC.